base/stacktraces.jl

```diff
@@ -255,11 +255,23 @@ function show_spec_linfo(io::IO, frame::StackFrame)
             color = get(io, :color, false) && get(io, :backtrace, false) ?
                         Base.stackframe_function_color() :
                         :nothing
-            printstyled(io, string(frame.func), color=color)
+            printstyled(io, Base.demangle_function_name(string(frame.func)), color=color)
         end
     elseif frame.linfo isa Core.MethodInstance
-        if isa(frame.linfo.def, Method)
-            Base.show_tuple_as_call(io, frame.linfo.def.name, frame.linfo.specTypes)
+        def = frame.linfo.def
+        if isa(def, Method)
+            sig = frame.linfo.specTypes
+            if def.nkw > 0
+                # rearrange call kw_impl(kw_args..., func, pos_args...) to func(pos_args...)
+                kwarg_types = Any[ fieldtype(sig, i) for i = 2:(1+def.nkw) ]
+                uw = Base.unwrap_unionall(sig)
+                pos_sig = Base.rewrap_unionall(Tuple{uw.parameters[(def.nkw+2):end]...}, sig)
+                orig = Base._which(pos_sig)
```

Comment:
This seems like probably the wrong method lookup. But also seems pretty awful that you're trying to decode the original method so that `kwarg_decl` can try to find `def` again—maybe just split that function so you can pass in `def`?

Or just inline the relevant line here (`kwnames = ccall(:jl_uncompress_argnames, Vector{Any}, (Any,), def.slot_syms)[2:(def.nkw + 1)]`)

Reply:
Note too that `Base.kwarg_decl` may resorts the list of argument names also (moves `vararg...` out of order), so using it with `zip` appears to be wrong (according to the comment in that function) for vararg methods.